backend/common/src/assign_id.rs

```diff
@@ -73,7 +73,10 @@ where
     }
 
     pub fn clear(&mut self) {
-        *self = AssignId::new();
+        // Leave the `current_id` as-is. Why? To avoid reusing IDs and risking
+        // race conditions where old messages can accidentally screw with new nodes
+        // that have been assigned the same ID.
+        self.mapping = BiMap::new();
```

Comment:
hmm, I don't understand.

Before we were resetting the `this.id == 0` and now just ignore this id and the next one will be `this.id +1`?
It looks like the `id` will wrap `if id + 1 > MAX` which will wrap in release mode IIRC.

Can you add an explicit panic or something so we are sure that it doesn't wrap around and ends up to replace an "old ID"?

Reply:
Yup, I did this because I worried that "old" IDs might be reused in some sorts of obscure race conditions, leading to new nodes being affected by "old" messages or something like that.

Indeed, eventually it'll overflow and panic, but since the ID is a 64 bit value it'll never realisically hit `usize::MAX` so I wasn't worried about overflowing (I almost considered adding an explicit wrapping_add but meh)

Reply:
Since it was so easy to do I did make the thing wrapping_add :)